lib/docker.js

```diff
@@ -129,15 +137,25 @@ DockerModule.prototype.registerContainer = function (src, containerId, opts) {
   });
 };
 
+function isMatchingImageName(targetName, imageName) {
+    var regex = new RegExp("^" + targetName + "$");
+    return regex.test(imageName);
+}
+
 function containerPort(dolphin, containerId) {
   return dolphin.containers.inspect(containerId).then(function (container) {
     var port = Object.keys(container.NetworkSettings.Ports)[0].split('/')[0];
-    var ip = container.NetworkSettings.IPAddress;
-    if (port && ip) {
-      return 'http://' + ip + ':' + port;
+
+    var netNames = Object.keys(container.NetworkSettings.Networks);
+    if (netNames.length === 1) {
+      var ip = container.NetworkSettings.Networks[netNames[0]].IPAddress;
+      if (port && ip) {
+        return 'http://' + ip + ':' + port;
+      }
     } else {
-      throw Error('No valid address or port ' + container.IPAddress + ':' + port);
+      //TODO: Implements opts for manually choosing the network/ip/port...
```

Comment:
ok, I'll add the network option. In case there are more than 1 networks, if no option is passed, an error will be thrown indicating that the network id must be provided when registering this container.